src/staff.js

```diff
@@ -9,6 +10,125 @@ const router = express.Router();
 router.use(authenticationMiddleware);
 router.use(authorizationMiddleware);
 
+router.get("/guests/:id/passes", getPass);
+router.post("/guests/:id/passes", postPass);
+router.post("/guests", postUser);
+
+async function getPassById(id) {
+  let passes;
+  try {
+    [passes] = await conn.query(
+      `SELECT * FROM pass  
+      WHERE userID = ?
+      ORDER BY dateOfBuying DESC`,
+
+      [id]
+    );
+  } catch (err) {
+    return err;
+  }
+  return passes;
+}
+
+async function getExpirationDate(startDate) {
+  let expirationDate;
+
+  try {
+    expirationDate = await conn.query(
+      `
+          SELECT DATE_ADD(?, INTERVAL 30 DAY);
+        `,
+      [startDate]
+    );
+  } catch (err) {
+    return next(err);
+  }
+  return expirationDate;
+}
+
+async function getPass(req, res, next) {
+  const ID = parseInt(req.params.id);
+
+  const passes = await getPassById(ID);
+
+  return res.status(200).send(passes);
+}
+
+async function getUserByEmail(email) {
+  const [users] = await conn.query("SELECT id FROM user WHERE email = ?", [
+    email,
+  ]);
+  return users[0];
+} // helyette unique?
+
+async function postUser(req, res, next) {
+  const { firstName, lastName, email, phoneNumber } = req.body;
+  let data;
+
+  if (!firstName || !lastName || !email) {
```

Comment:
This is very basic validation. I’d recommend googling a validation library that has built-in validator for email and phone number validity should also be checked.